src/components/CommentPost.jsx

```diff
@@ -0,0 +1,86 @@
+import { useContext, useState } from "react"
+import LoggedInContext from "../contexts/Logged-In-User-Context"
+import { postCommentOnGivenArticleId } from "../utils/POST-comment"
+import { commentValidator } from "../utils/comment-validator"
+
+const PostCommentField = (props) => {
+    
+    const currentArticleId = props.currentArticleId
+    const currentComments = props.currentComments
+    const setCurrentComments = props.setCurrentComments
+
+    const { loggedInUser, setLoggedInUser } = useContext(LoggedInContext)
+    const [newCommentAttempt, setNewCommentAttempt] = useState('')
+    const [commentPosted, setCommentPosted] = useState(false)
+    const [isCommentInvalid, setIsCommentInvalid] = useState(false)
+    const [postError, setPostError] = useState(false)
+    
+    const handleSubmit = (event) => {
+        event.preventDefault()
+        if (commentValidator(newCommentAttempt)){
+            setIsCommentInvalid(false)
+            const optimisticComment = {
```

Comment:
this looks great! But we probably don't need to add the comment id as we aren't rendering this information anywhere and it will most likely be incorrect! - also we will end up having multiple comments with the same 'key' when they are mapped through - in this case you can just use the index number of the comment in the array.